app/utils/index.ts

```diff
@@ -0,0 +1,18 @@
+class Generator {
+  private *idGenerator(start: number, end: number, step: number) {
+    for (let i = start; i < end; i += step) {
+      yield i;
+    }
+  }
+  private newGen = this.idGenerator(this.start, this.end, this.step);
```

Comment:
I am still unclear about the side effects, if any, of not having this in the constructor. Perhaps we need to test by instantiating 2 generators and asserting that there there is no side effect.